src/backend/auth.api.ts

```diff
@@ -143,18 +143,20 @@ const getUserByUserId = async (userId: string) => {
 };
 
 const login = async (email: string, password: string) => {
+  // Delete current session, if any
   try {
-    await account.deleteSessions();
-    const response = await account.createEmailPasswordSession(email, password);
-
-    if (!response) {
+    await account.deleteSession("current");
+  } catch (error: any) {
+    if (error?.type !== "general_unauthorized_scope") {
       throw new Error("Login failed");
     }
+  }
 
-    const accountId: string = response && response.userId;
-
+  try {
+    const session = await account.createEmailPasswordSession(email, password);
+    const { userId: accountId, expire } = session;
     const user = await getUserByAccountId(accountId);
-    return user;
+    return { user, expires: new Date(expire) };
```

Comment:
this will expire immediately, you need to add some buffer time. I prefer to add 2 days

Reply:
I would like to know why would this expire immediately?

As per the [Appwrite docs](https://appwrite.io/docs/references/cloud/models/session), the `session` object contains an `expire` property which is a date string. This string is converted to a date object here because it is required as such by nookies.

Also, the cookie expire time is set to be in sync with the appwrite session validation (1 year by default), that can be changed.

**Cookie expire time (matches appwrite session length):**
![image](https://github.com/Sanchitbajaj02/palettegram/assets/80148777/2ac1627f-cf62-45c6-8b34-32a537c9e96b)

**Appwrite console to set session length (auth > security):**
![image](https://github.com/Sanchitbajaj02/palettegram/assets/80148777/0e8a442a-e3ab-48c9-b4cd-94ec887fdc26)
Changing this to 2 days would set the cookie expire time to 2 days for users.

Reply:
Let me test that out locally, and then finalize the PR